modules/run-consul/run-consul

```diff
@@ -656,11 +318,59 @@ function run {
       "$redundancy_zone_tag" \
       "$disable_upgrade_migration" \
       "$upgrade_version_tag" \
+      "$enable_acl"
       "${recursors[@]}"
   fi
 
+  log_info "Creating Consul systemd file"
   generate_systemd_config "$SYSTEMD_CONFIG_PATH" "$config_dir" "$data_dir" "$systemd_stdout" "$systemd_stderr" "$bin_dir" "$user" "${environment[@]}"
+  log_info "Starting Consul"
   start_consul
+
+  if [[ "$enable_acl" == "true" ]]; then
+    local -r asg_name=$(aws_wrapper_get_asg_name $MAX_RETRIES $SLEEP_BETWEEN_RETRIES_SEC)
```

Comment:
sorry one other question - and i have low knowledge of bash so i don't know if this is plausible.

i have an equivalent set of code (up to 364), basically using vault rather than ssm for the token storage.

i can see that you pass the "ssm" parameter so other stores could be added in the future.  but each of those functions is then going to have a switch statement in it.  i was wondering whether it is straightforward to just load the functions used based on a parameter.  Something like:

if [[ "$enable_acl" == "true" ]]; then
   if [[ "$acl_provider_path" != "" ]]; then
     source $acl_provider_path;
   else
     source ./aws-ssm-bootstrapper
   fi
fi

providing a name rather than a path would also work - other than ssm and vault, i can't think of other common things one might want to do, but i guess you can't rule out some kind of nfs share or other non-aws-native route

Reply:
Hmm... so the idea your suggesting is that the `run-consul` script have a path for the acl storage provider, source that file dynamically, then call some known function name (like the current `write_acl_token` one)? It's certainly doable, I just get a little antsy about dynamically sourcing some random file, since with bash you can't really do things like enforce interfaces. The current idea is that the `write_acl_token` function is the entrypoint for writing tokens to durable storage, and that it would be enhanced over time to support multiple ones (like Vault, S3, etc) as needed.

Reply:
Oh and should also note that, once another provider is required, the `run-consul` script would also be enhanced to have a parameter named something like `--acl-storage-type`, which would then be passed in to the `write_acl_token` function (vs right now where it's hard coded to pass 'ssm').

Reply:
I've ping'd you a PR and comment, so you had a diff to look at rather than my unclear explanations

Reply:
Thanks for the PR, that definitely helped me understand. Let me see what I can do there as I do agree it's a cleaner approach.